modules/widgets/milestone/milestone.php

```diff
@@ -0,0 +1,360 @@
+				'<span class="difference">' . esc_html( $number ) . '</span>',
+				'<span class="label">' . esc_html( $label ) . '</span>'
+			) . '</div>';
+
+			self::$config_js['instances'][] = array(
+				'id'      => $args['widget_id'],
+				'diff'    => $diff,
+				'message' => $instance['message'],
+			);
+		}
+
+		echo '</div><!--milestone-content-->';
+
+		echo $args['after_widget'];
+		stats_extra( 'widget_view', 'milestone' );
+    }
+
+    /**
+     * Update
+     */
+    function update( $new_instance, $old_instance ) {
+		return $this->sanitize_instance( $new_instance );
+    }
+
+	/*
+	 * Make sure that a number is within a certain range.
+	 * If the number is too small it will become the possible lowest value.
+	 * If the number is too large it will become the possible highest value.
+	 *
+	 * @param int $n The number to check.
+	 * @param int $floor The lowest possible value.
+	 * @param int $ceil The highest possible value.
+	 */
+	function sanitize_range( $n, $floor, $ceil ) {
+		$n = (int) $n;
+		if ( $n < $floor )
+			$n = $floor;
+		else if ( $n > $ceil )
+			$n = $ceil;
+		return $n;
+	}
+
+	/*
+	 * Sanitize an instance of this widget.
+	 *
+	 * Date ranges match the documentation for mktime in the php manual.
+	 * @see http://php.net/manual/en/function.mktime.php#refsect1-function.mktime-parameters
+	 *
+	 * @uses Milestone_Widget::sanitize_range().
+	 */
+	function sanitize_instance( $dirty ) {
+		$now = (int) current_time( 'timestamp' );
+
+		$dirty = wp_parse_args( $dirty, array(
+			'title'   => '',
+			'event'   => __( 'The Big Day' ),
+			'message' => __( 'The big day is here.' ),
+			'day'     => date( 'd', $now ),
+			'month'   => date( 'm', $now ),
+			'year'    => date( 'Y', $now ),
+			'hour'    => 0,
+			'min'     => 0,
+		) );
+
+		$allowed_tags = array(
+			'a'      => array( 'title' => array(), 'href' => array() ),
+			'em'     => array( 'title' => array() ),
+			'strong' => array( 'title' => array() ),
+		);
+
+		$clean = array(
+			'title'   => trim( strip_tags( stripslashes( $dirty['title'] ) ) ),
+			'event'   => trim( strip_tags( stripslashes( $dirty['event'] ) ) ),
+			'message' => wp_kses( $dirty['message'], $allowed_tags ),
+			'year'    => $this->sanitize_range( $dirty['year'],  1901, 2037 ),
+			'month'   => $this->sanitize_range( $dirty['month'], 1, 12 ),
+			'hour'    => $this->sanitize_range( $dirty['hour'],  0, 23 ),
+			'min'     => zeroise( $this->sanitize_range( $dirty['min'], 0, 59 ), 2 ),
+		);
+
+		$clean['day'] = $this->sanitize_range( $dirty['day'], 1, date( 't', mktime( 0, 0, 0, $clean['month'], 1, $clean['year'] ) ) );
+
+		return $clean;
+	}
+
+    /**
+     * Form
+     */
+    function form( $instance ) {
+		$instance = $this->sanitize_instance( $instance );
+        ?>
+
+	<div class="milestone-widget">
+        <p>
+        	<label for="<?php echo $this->get_field_id( 'title' ); ?>"><?php _e( 'Title' ); ?></label>
```

Comment:
Doesn't hurt to err on the side of caution and `esc_attr()` the result of `get_field_id` and `get_field_name`